Add AreaEffect prototype and adjust arena/constants

ngbrown opened the pull request:
I added the `AreaEffect` which was needed for Collect and Control advanced (to avoid the damage effects). I also adjusted the `arena/constants` to be split types/const like the other constants.

Also, adjusted ScoreCollector to extend GameObject like in the docs. Just because it has a `my` property, I don't think it should derive from owned because the control flips based on score.

thmsndk commented:
🤔 towers can be neutral in advanced CTF can't they? can the ScoreCollector be neutral as well? 
because then we might need to update .my to also be undefined e.g. `my?: boolean` 
I can see we need to do that for towers at least

thmsndk commented on dist/game/constants.d.ts:
```diff
@@ -181,9 +181,10 @@ declare module "game/constants" {
+  export const RESOURCE_ENERGY: RESOURCE_ENERGY;
 
-  export type ResourceConstant = typeof RESOURCE_ENERGY | typeof RESOURCE_SCORE;
+  export type ResourceConstant = RESOURCE_ENERGY | RESOURCE_SCORE | RESOURCE_SCORE_X | RESOURCE_SCORE_Y | RESOURCE_SCORE_Z;
```
We might want to invent a `ARENA_RESOURCE` union type we can import here in game, so it can be specified in the arena module?

ngbrown replied:
To be consistent with the other union types, I added `ArenaResourceConstant`.

thmsndk reviewed (changes requested):
Thanks for the update @ngbrown could you write the testcase for this in `dist/test/screeps-arena-tests.ts` and verify that `npm test` does not complain about the usage of the typing?

ngbrown commented:
I added tests and rebased.

Regarding towers being neutral, maybe my argument ownership changing was weak, but the help documentation specifically said "class StructureTower Extends OwnedStructure" and "class ScoreCollector Extends GameObject".
